### ISSUE TITLE
Sitemap hreflang syntax invalid for regional language variants

### ISSUE THREAD

**Issue by strophy** [rtfd/readthedocs.org#5446]
The `hreflang` for regional variations in `sitemap.xml` follows the format of the URL language slug [generated by Sphinx](http://www.sphinx-doc.org/en/master/usage/configuration.html#confval-language), e.g. `zh_CN` for Chinese (China). This is invalid syntax for `hreflang` in a sitemap ([reference](https://en.wikipedia.org/wiki/Hreflang#Common_Mistakes)), where a hyphen must be used instead, e.g. `zh-CN`. Alternatively, it is also valid to define the script of the language instead, e.g. `zh-Hans` for Simplified Chinese. Some sort of library to perform this conversion, a lookup table or a user-facing option could help fix this.

Example from: https://docs.dash.org/sitemap.xml
```
<xhtml:link rel="alternate" hreflang="zh_CN" href="http://docs.dash.org/zh_CN/latest/"/>
```

**Comment by stsewd** [comment]
The fact that we rely on sphinx for this make it hard to follow the standard way, there are other parts of the code that we should take care of too.

**Comment by saadmk11** [comment]
Hey @humitos can you tell me how should this be approached? And what needs to be changed to fix this?

**Comment by humitos** [comment]
@saadmk11 the `hreflang` label from the sitemap uses some incorrect codes. Those incorrect ones (for example, `zh_CN`) should be replaced for the correct ones (`zh-CN`). The linked document from the description has more detailed information.
